### ISSUE TITLE
MAP_FRAME_PEN auto-scaling is not ideal in modern mode

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#6786]
**Description of the problem**

The following script plots three basemaps with different sizes (10c, 5c and 1c).
```
gmt begin map
gmt basemap -R0/10/0/10 -JX10c -B0
gmt colorbar -Crainbow -Baf

gmt basemap -R0/10/0/10 -JX5c -B0 -X10c
gmt colorbar -Crainbow -Baf

gmt basemap -R0/10/0/10 -JX1c -B0 -X5c
gmt colorbar -Crainbow -Baf
gmt end show
```
Here is the output image. As you can see, the frame pen thickness is almost identical (with very small thickness changes). The default pen thickness looks good for a 15-cm-wide basemap, but is too thick for a 5-cm-wide or 1-cm-wide basemap.

![map](https://user-images.githubusercontent.com/3974108/173028013-13287b5a-6722-472f-a7bd-2a1f8440aa9b.png)

**System information**

* GMT version (`gmt --version`): `6.4.0_a7493e9_2022.05.26`

**Comment by PaulWessel** [comment]
Yes, it is not ideal, but I guess the auto-scaling was set up to get _reasonable_ annotation font sizes for reasonable sizes.  1 cm maps are **not reasonable** and ends up with too large annotations and pen thicknesses.  Given our font size scaling

`fontsize = (2.0/15.0) * (map_dim_cm - 10.0) + 9;	/* Gives result in points for plot dimension in cm */`

and overall scale for thinkgs like pen widths:

`scale = fontsize / 10.0;	/* scaling for offsets, pen widths and lengths normalized to the modern 10p size */`

the 1 cm map only yields a scale reduction of 0.78.

**Comment by seisman** [comment]
I agree that 1-cm-wide maps are not reasonable.

I feel that the pen thickness is still a little too thick for the colorbar for the 5-cm-wide maps. Compared to the 10-cm-wide maps, the height of the colorbar has a scale reduction of 0.5, but the colorbar frame pen thickness has a scale reduction of 0.83. I think that's why I have to change the frame pen thickness for most colorbars in my own plots.

**Comment by PaulWessel** [comment]
I agree we need to do something about the colorbar pen.

**Comment by PaulWessel** [comment]
Because the thickness of the bar is usually very small compared to its length, it is not helpful to use the same pen width as for the map.  Even if we reduced it by the fractional length to map with (e.g. 0.8) it is still to think.  Perhaps it is reasonable to simply use a factor of 0.5 * framepen->width for colorbars where framepen is the width scaled for the size of the map?

**Comment by PaulWessel** [comment]
BTW, do you get this now after the cmocean merge?

```
gmt grdimage @earth_relief_01d -R0/30/0/40 -JM5c -B -png t
grdimage [ERROR]: File /Users/pwessel/UH/RESEARCH/CVSPROJECTS/GMTdev/gmt-dev/share/geo.cpt was not found
```

**Comment by PaulWessel** [comment]
> BTW, do you get this now after the cmocean merge?
> 
> ```
> gmt grdimage @earth_relief_01d -R0/30/0/40 -JM5c -B -png t
> grdimage [ERROR]: File /Users/pwessel/UH/RESEARCH/CVSPROJECTS/GMTdev/gmt-dev/share/geo.cpt was not found
> ```

Never mind, went away after rebuild.
